### PR TITLE
add nrow to select

### PR DESCRIPTION
**Pull request by bkamins**
Allow an easy way to get a number of rows in `select`. It was such a clean idea that I could not resist it (but maybe other people will think otherwise).

I think having this as a special case is justified.

When this is merged the full benefit will be in https://github.com/JuliaData/DataFrames.jl/pull/2158, as `nrow` is routinely required in split-apply-combine operations and now we did not have a clean and quick way to do it.

### CODE REVIEW

File: src/abstractdataframe/selection.jl [hunk 1]
```diff
@@ -198,6 +205,10 @@ using `_`; if more than three columns are passed then the name consists of the
 first two names and `etc` suffix then, e.g. `[:a,:b,:c,:d] => fun` produces
 the new column name `:a_b_etc_fun`.
 
+As a special rule passing `nrow` as an argument creates a column named `:nrow`
+containing a number of rows in a source data frame, and passing `nrow => new_column_name`
+stores a number of rows in a source data frame in `new_column_name` column.
```
Comment (nalimilan) [inline src/abstractdataframe/selection.jl]:
```suggestion
stores the number of rows in source data frame in `new_column_name` column.
```

File: src/abstractdataframe/selection.jl [hunk 2]
```diff
@@ -338,6 +349,10 @@ using `_`; if more than three columns are passed then the name consists of the
...
```
Comment (nalimilan) [inline src/abstractdataframe/selection.jl]:
```suggestion
stores the number of rows in source data frame in `new_column_name` column.
```

File: src/abstractdataframe/selection.jl [hunk 3]
```diff
@@ -198,6 +205,10 @@ using `_`; if more than three columns are passed then the name consists of the
 first two names and `etc` suffix then, e.g. `[:a,:b,:c,:d] => fun` produces
 the new column name `:a_b_etc_fun`.
 
+As a special rule passing `nrow` as an argument creates a column named `:nrow`
```
Comment (nalimilan) [inline src/abstractdataframe/selection.jl]: Maybe that's more explicit?
```suggestion
As a special rule passing `nrow` without specifying `old_column` creates a column named `:nrow`
```

**Comment by bkamins** [comment]
Thank you. I will integrate it to other PRs now.

**Pull request merged**